### ISSUE TITLE
Downloaded Foxx services are not valid zip files

### ISSUE THREAD

**Issue by andybarilla** [arangodb/arangodb#1883]
I tried downloading a Foxx service, one of my own and a premade one (itzpapalotl), the zip files can not be read as valid zip files. 

```
$ Downloads: unzip itz_1.2.0.zip 
Archive:  itz_1.2.0.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of itz_1.2.0.zip or
        itz_1.2.0.zip.zip, and cannot find itz_1.2.0.zip.ZIP, period.
```

Also, when trying to 'Replace' with the same downloaded zip the following error occurs:

```
2016-06-20T21:13:31Z [24167] ERROR ArangoError: <path> must be a valid directory name (have '/tmp/arangod_AMxUqz/apps/tmp-24167-130558203')
2016-06-20T21:13:31Z [24167] ERROR     at Error (native)
2016-06-20T21:13:31Z [24167] ERROR     at _validateService (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/manager.js:1067:8)
2016-06-20T21:13:31Z [24167] ERROR     at Object.upgrade (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/manager.js:1386:3)
2016-06-20T21:13:31Z [24167] ERROR     at Middleware.<anonymous> (/usr/share/arangodb3/js/apps/system/_admin/aardvark/APP/foxxes.js:90:29)
2016-06-20T21:13:31Z [24167] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:388:15)
2016-06-20T21:13:31Z [24167] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:384:7)
2016-06-20T21:13:31Z [24167] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:384:7)
2016-06-20T21:13:31Z [24167] ERROR     at /usr/share/arangodb3/js/node/node_modules/lodash/index.js:7363:25
2016-06-20T21:13:31Z [24167] ERROR     at Middleware.router.use (/usr/share/arangodb3/js/apps/system/_admin/aardvark/APP/foxxes.js:51:3)
2016-06-20T21:13:31Z [24167] ERROR     at next (/usr/share/arangodb3/js/server/modules/@arangodb/foxx/router/tree.js:388:15)
```

**Comment by pluma** [comment]
@dothebart I think you said you're looking into this? Give me a shout if there's any way I can help you with this.

**Comment by jsteemann** [comment]
Reproduced.

**Comment by jsteemann** [comment]
Just had a look into this. The server response seems correct. At least the server sends the zip file with correct contents and correct `Content-Length` value. The response has a `Content-Type` of `application/zip`, which is appropriate (changing it to `application/octet-stream` doesn't help either). `Content-Disposition` is `attachment; filename="..."` which is also ok.
I can also download the file correctly when calling the URL via curl from the command-line.

So I guess the problem occurs because the download is triggered from the web frontend, with some AJAX calls. This seems to wrap the correct server response into a JavaScript string, so it gets broken there.
@pluma: I think I am not able to fix this any soon, can you take over?

**Comment by jsteemann** [comment]
For completeness: downloads of Foxx apps don't work in Firefox at all. In Chrome download the (broken) zip file, in Firefox nothing seems to happen though the download URL is called.

**Comment by pluma** [comment]
I have an idea how to fix this. I'll look into it.

**Comment by jsteemann** [comment]
@pluma thanks!

**Comment by pluma** [comment]
This has been fixed in devel and 3.0.
